ext/liquid_c/block.c

```diff
@@ -68,7 +75,23 @@ static VALUE rb_block_parse(VALUE self, VALUE tokens, VALUE options)
             }
             case TOKEN_VARIABLE:
             {
-                VALUE args[2] = {rb_enc_str_new(token.str + 2, token.length - 4, utf8_encoding), options};
+                const char *start = token.str + 2;
+                long length = token.length - 4;
+
+                if (token.str[2] == '-') {
+                    if (RARRAY_LEN(nodelist) > 0) {
+                        VALUE last_node = RARRAY_AREF(nodelist, RARRAY_LEN(nodelist)-1);
+                        rb_funcall(last_node, intern_rstrip, 0);
+                    }
+                    start++;
+                    length--;
+                }
+
+                if (token.str[token.length - 3] == '-') {
```

Comment:
I would use `if (length > 0 && start[length - 1] == '-') {` because I am concerned about a case like `{{-}}` would cause length to go negative, which is a scary state to be in C code.